app/ui/src/app/integrations/edit-page/step-configure/step-configure.component.ts

```diff
@@ -224,12 +224,14 @@ export class IntegrationsStepConfigureComponent extends FlowPage implements OnIn
       category
     );
     */
-    // supress null values
-    Object.keys(values).forEach(key => {
-      if (values[key] === 'null') {
-        values[key] = undefined;
+    if (values) {
+      // supress null values
```

Comment:
I think the intent is pretty obvious and we don't need a comment to clarify it LOL :D 

BTW, do you want to assign the `undefined` value to these fields or just remove those from the `values` object (provided they've been flagged as optional in its TypeScript interface)? If you strive to do the latter, I'd suggest to [take a look into this](https://github.com/syndesisio/syndesis/blob/7e4370a6c3af492054771ce88c80ad67c63dbfc9/app/ui/src/app/customizations/api-connector/api-connector.service.ts#L80-L94).

In the example provided within our codebase the _for/in_ loop is constrained to remove only the empty/null/undefined properties found in the object definition that belong to its type, not including those which are inherited though. 

Aside from the comment and this word of caution, it LGTM

Reply:
Yeah, so this apparently came about as a side effect of the backend setting values to null for empty strings.  I considered actually putting this right into the form service, though it does bug me that basically then you can never have a value in a `configuredProperties` map set to the string `null` :-)

I'm fine with setting it to just `undefined`, when the form service builds the form the resulting value will either be unset or whatever the default value is set to in the properties object.